get-charm-paths/src/get_charm_paths.py

```diff
@@ -0,0 +1,56 @@
+#!/bin/env python
+import argparse
+from pathlib import Path
+
+
+OUTPUT_VARIABLE_NAME = "charm_paths"
+
+
+def find_charms_in_dir(base_dir: str, charms_subdir="charms"):
+	"""Returns a list of paths to charm directories in base_dir.
+
+	Searches for:
+	* any subdirectories in base_dir/charms_subdir/ are charm directories
+	* if no charm dirs are found, checks if base_dir is itself a charm directory
+	* if still nothing is found, returns []
+
+	Directories are identified as charm directories by the presence of a "metadata.yaml" file.
+	Returned paths include base_dir and charms_subdir in their path, as applicable.
+	"""
+	base_dir = Path(base_dir)
+	charms_dir = base_dir / charms_subdir
+	metadata_files = list(charms_dir.glob("*/metadata.yaml"))
```

Comment:
I think this will return an empty list which will make `charm_dirs` also an empty list, which after you pass to `stringify_paths()` will just be `/`, which can be problematic (?)

Reply:
To make sure I understand what you're raising, are you saying that:
* `list(charms_dir.glob("*/metadata.yaml"))` may return an empty list (`[]`)
* that results in `charm_dirs=[]`
* which results in `stringify_paths(charm_dirs)` being fed an empty list `[]` as input, leading to a return of `charm_dirs='/'`
?

Reply:
I added some extra README.md content which I think addresses this, and fleshed out the tests a bit to make it clearer what happens.  In the case described above, `stringify_paths([])` yields the output of an empty list `[]`, which leads to the script emitting the output of an empty list as well rather than an error.  This would be equivalent to "no charms found".

Reply:
Got it.